parse.py

```diff
@@ -0,0 +1,292 @@
+#-*- coding: utf-8 -*-
+
+import html5, string
+
+# Global variables
+_tags = None
+_domParser = None
+
+def _convertEncodedText(txt):
+	"""
+	Convert HTML-encoded text into decoded string.
+
+	The reason for this function is the handling of HTML entities, which is not
+	properly supported by native JavaScript.
+
+	We use the browser's DOM parser to to this, according to
+	https://stackoverflow.com/questions/3700326/decode-amp-back-to-in-javascript
+
+	:param txt: The encoded text.
+	:return: The decoded text.
+	"""
+	global _domParser
+
+	if _domParser is None:
+		_domParser = eval("new DOMParser")
+
+	dom = _domParser.parseFromString("<!doctype html><body>" + str(txt), "text/html")
+	return dom.body.textContent
+
+def _buildDescription():
+	"""
+	Generates a dictionary of all to the html5-library
+	known tags and their associated objects and attributes.
+	"""
+	tags = {}
+
+	for cname in dir(html5):
+		cl = getattr(html5, cname)
+		try:
+
+			inst = cl()
+			if cname != "Body" and isinstance(inst, html5.Widget):
+				attr = []
+
+				for fname in dir(inst):
+					if fname.startswith("_set"):
+						attr.append(fname[4:].lower())
+
+				tags[cname.lower()] = (cl, attr)
+
+		except:
+			pass
+
+	for tag in sorted(tags.keys()):
+		print("%s: %s" % (tag, ", ".join(sorted(tags[tag][1]))))
+
+	return tags
+
+def fromHTML(html, appendTo = None, bindTo = None):
```

Comment:
General Question: Why parsing yourself? Browsers can do: https://developer.mozilla.org/en-US/docs/Web/API/DOMParser

Reply:
Maybe this can be replaced later. Currently, our HTML parser should only understand tags and attributes supported by our html5 library, to make app development easier, faster and more reliable.